README.md

```diff
@@ -85,6 +84,53 @@ function worker (arg, cb) {
 }
 ```
 
+### Using with TypeScript (callback API)
+
+```ts
+'use strict'
+
+import * as fastq from "fastq";
+import type { queue, done } from "fastq";
+
+type Task = {
+  id: number  
+}
+
+const q: queue<Task> = fastq(worker, 1)
+
+q.push({ id: 42})
+
+function worker (arg: Task, cb: done) {
+  console.log(arg.id)
+  cb(null)
+}
+```
+
+### Using with TypeScript (promise API)
+
+```ts
+'use strict'
+
+import * as fastq from "fastq";
+import type { queueAsPromised } from "fastq";
+
+type Task = {
+  id: number
+}
+
+const q: queueAsPromised<Task> = fastq.promise(asyncWorker, 1)
+
+q.push({ id: 42})
+
+async function asyncWorker (arg: Task): Promise<void> {
+  try {
```

Comment:
@mcollina Do you prefer try-catch or Promise.resolve().then().catch() here? Is it generally safe not to catch a promise rejection here, will that end up as an unhandled rejection or fastq handles that itself?

Reply:
This is catched by fastq, there is no need to add a try-catch here: https://github.com/mcollina/fastq/blob/218eac7cb0a7c65cf4a81c3c370d269d29ac7a2b/queue.js#L216.

Reply:
Updated!